File: c100247. Maximize Happiness of Selected Children.py
def maximum_happiness_sum(happiness, k):
    n = len(happiness)
    if k >= n:
        return sum(happiness)
    
    happiness.sort(reverse=True)
    total_happiness = sum(happiness[:k])
    
    decrement = 0
    for i in range(1, k):
        if happiness[i] == happiness[i - 1]:
            decrement += i
    
    return total_happiness - decrement

happiness1 = [1,2,3]
k1 = 2
print(maximum_happiness_sum(happiness1, k1)) 

happiness2 = [1,1,1,1]
k2 = 2
print(maximum_happiness_sum(happiness2, k2))  

happiness3 = [2,3,4,5]
k3 = 1
print(maximum_happiness_sum(happiness3, k3)) 

File: c100247. Maximize Happiness of Selected Children.rb
def maximum_happiness_sum(happiness, k)
  n = happiness.length
  return happiness.sum if k >= n
  
  happiness.sort!.reverse!
  total_happiness = happiness[0, k].sum
  
  decrement = (1..k-1).sum
  total_happiness - decrement
end

# Test cases
happiness1 = [1,2,3]
k1 = 2
puts maximum_happiness_sum(happiness1, k1)  # Output: 4

happiness2 = [1,1,1,1]
k2 = 2
puts maximum_happiness_sum(happiness2, k2)  # Output: 1

happiness3 = [2,3,4,5]
k3 = 1
puts maximum_happiness_sum(happiness3, k3)  # Output: 5
